packages/react-button/src/components/Button/SPEC.md

```diff
@@ -0,0 +1,297 @@
+
+    /**
+     * A button supports different sizes.
+     * @default 'medium'
+     */
+    size?: 'small' | 'medium' | 'large';
+
+    /**
+     * A button can be styled to blend into its background and become less emphasized.
+     * @default false
+     */
+    subtle?: boolean;
+
+    /**
+     * A button can be styled such that it has no background or border styling and is just emphasized through its
+     * content styling.
+     * @default false
+     */
+    transparent?: boolean;
+  };
+```
+
+### Styling Tokens
+
+TBD once we decide on component tokens work.
+
+## Structure
+
+For `Buttons` rendering as `<button>`:
+
+_Icon before children:_
+
+```tsx
+<button class="root" href={href}>
+  <span class="loader" />
+  <span class="icon" />
+  {children}
+</button>
+```
+
+_Icon after children:_
+
+```tsx
+<button class="root" href={href}>
+  <span class="loader" />
+  {children}
+  <span class="icon" />
+</button>
+```
+
+For `Buttons` rendering as anything other than `<button>` (the example below uses `<div>`):
+
+```tsx
+<div class="root" href={href}>
+  <span class="loader" />
+  <span class="icon" />
+  {children}
+</div>
+```
+
+_Icon after children:_
+
+```tsx
+<div class="root" href={href}>
+  <span class="loader" />
+  {children}
+  <span class="icon" />
+</div>
+```
+
+## Migration
+
+See [MIGRATION.md](https://github.com/microsoft/fluentui/blob/master/packages/react-button/MIGRATION.md).
+
+## Behaviors
+
+### States
+
+The following section describes the different states in which a `Button` can be throughout the course of interaction with it.
+
+#### Enabled state
+
+An enabled `Button` communicates interaction by having styling that invites the user to click/tap on it to trigger an action.
+
+#### Disabled state
+
+A disabled `Button` is non-interactive, disallowing the user to click/tap on it to trigger an action.
+
+#### Hovered state
+
+A hovered `Button` changes styling to communicate that the user has placed a cursor above it.
+
+#### Focused state
+
+A focused `Button` changes styling to communicate that the user has placed keyboard focus on it. This styling is usually the same to the one in the hovered state plus extra styling on the outline to indicate keyboard focus has been placed on the component.
+
+#### Pressed state
+
+A pressed `Button` changes styling to communicate that the user is currently pressing it.
+
+#### Loading state
+
+A loading `Button` renders a `loader` before all the other content to indicate that it is waiting for another action before allowing itself to be interacted with.
+
+### Interaction
+
+#### Keyboard interaction
+
+The following is a set of keys that interact with the `Button` component:
+
+| Key                      | Description                            |
+| ------------------------ | -------------------------------------- |
+| `Enter`                  | Executes the `Button` action.          |
+| `Space`                  | Executes the `Button` action.          |
```

Comment:
Investigate how `Enter` and `Space` work with native browser `button` and have the `Button` events follow the native events as close as possible.

Possibly want to check with conformance tests.

Open question: Do we really need to do specific checks for the `as` prop? We need to be clear if and when we deviate because this starts getting into magic handling of props which has other potential drawbacks such as bundle size increases.